### ISSUE TITLE
Z5 BBT export error with a postscript that works with a previous version

### ISSUE THREAD

**Issue by ajustine** [retorquere/zotero-better-bibtex#712]
Hi,

I am trying to export some references using a postscript that works very well with my previous version of Zotero and BBT, but does not work with Z5 and pre-release BBT 5.0.2.4102 ...

The report ID for the version that does not work is G9ENINYC.
I cannot send an error report for the version that works, but when I try to do so, I got a message speaking about a RequestId 1F38D2C8200C36BA (don't know if that can be of any help...)

The postscript I use is the following:
```
this.add({ name: 'x-audience', value: 'international' });
this.add({ name: 'x-popularlevel', value: 'no' });
this.add({ name: 'x-peerreviewing', value: 'yes'});

if (this.has.language) {
  this.has.language.name = 'x-language';
}

if (Translator.BetterBibTeX || Translator.BetterBibLaTeX) {
  this.add({ Archive: this.item.archive });
}
if (this.has.Archive) {
  this.has.Archive.name = 'x-onbehalfof';
}

if (!this.item.keywords) {
  this.add({ name: 'keywords', value: '[No keyword]'});
}

if (!this.item.abstract) {
  this.add({ name: 'abstract', value: '[No abstract]'});
}
```

My problem has obviously something to do with the 2 last items, but I can figure out why it works with my previous version of Zotero and BBT and not with the latest releases.

If you have a bit of time to spare to have a look at it, I would be very grateful.
Many thanks in advance !

**Comment by retorquere** [comment]
It's best to check for `this.has`(which checks the reference under construction) rather than `this.item` (which points to the internal representation of Zotero of the item, and some fields have changed since Z4; `this.item.abstract` is now `this.item.abstractNote` AFAICT), the last check finds no `abstract` because of the field name change, and tries to add it again. But BBT has checks on duplicate fields by default and fails on those. Those checks were mostly for me (to make sure my test suite would spot it if I added a duplicate field unintentionally), and that behaviour has leaked into the postscript.

I've lined up a build (4118) which will make that error a soft-fail, but you will likely want to change the last part to 

```
if (!this.has.keywords) { // this can't have worked before, as the Zotero item has "tags", not "keywords"
  this.add({ name: 'keywords', value: '[No keyword]'});
}

if (!this.has.abstract) {
  this.add({ name: 'abstract', value: '[No abstract]'});
}
```

Unfortunately there's no way for me to report errors from translator runs, so the only way for you to know what's going on is to restart with debug logging from the help menu and pour through the logs.
